File: services/graphql-ims/src/models/suppliers.rs
#[derive(Clone, Debug, PartialEq, Eq, async_graphql :: SimpleObject)]
pub struct Model {
    pub id: Uuid,
    pub name: String,
    pub contact_person: Option<String>,
    pub email: Option<String>,
    pub phone_number: Option<String>,
    pub created_at: Option<DateTime>,
    pub updated_at: Option<DateTime>,
}

File: src/graphql/queries/wms/suppliers.ts
import { graphql } from "@/lib/graphql/client";

/**
 * Query to fetch a single supplier by its ID.
 */
export const getSupplier = graphql(`
  query GetSupplier($id: UUID!) {
    wms {
      supplier(id: $id) {
        id
        name
        contactPerson
        email
        phoneNumber
        createdAt
        updatedAt
      }
    }
  }
`);

/**
 * Query to fetch a list of suppliers with pagination.
 */
export const getSuppliers = graphql(`
  query GetSuppliers($limit: Int!, $page: Int!) {
    wms {
      suppliers(limit: $limit, page: $page) {
        id
        name
        contactPerson
        email
        phoneNumber
        createdAt
        updatedAt
      }
    }
  }
`);
